model/annotations/task_annotations.go

```diff
@@ -97,6 +97,30 @@ func MoveSuspectedIssueToIssue(annotationId string, issue IssueLink, username st
 	)
 }
 
+func UpdateAnnotationNote(taskId string, execution int, originalMessage, newMessage, username string) error {
+	note := Note{
+		Message: newMessage,
+		Source:  &Source{Requester: UIRequester, Author: username, Time: time.Now()},
+	}
+
+	annotation, err := FindOneByTaskIdAndExecution(taskId, execution)
+	if err != nil {
+		return errors.Wrapf(err, "error finding task annotation")
+	}
+
+	if annotation != nil && annotation.Note != nil && annotation.Note.Message != originalMessage {
+		return errors.New("note has already been updated")
```

Comment:
I worry that "note has already been updated" might not make it clear to the user that it may have been updated by someone else.

Reply:
Yeah I can't decide how to word it. Is this better? Also from the UI, this error would come with a refreshed page I assume, and so the user would also see the new note with the new source and that would probably help clarify.

Reply:
Good point. And we can reevaluate the message when we build the UI for it.